### PR TITLE
Feat/build section

### LINKED ISSUE #248: Improve the Build section.
A lot of the contents in the `/build` section is good, but most of it has been written by multiple writers, making it disjointed and not very friendly to look at. We should improve the Build section by:

- [ ] Adding a purpose for each page.
- [ ] Adding further reading/resources to each page, where necessary.
- [x] Re-structure the `/build/examples` into single pages for clarity (and to make the sidebar nav easier to deal with).
- [x] Grammar and spell check _everything_.
- [ ] List redirects needed after moving things around.

### PR DESCRIPTION
**Pull request by johnnymatthews**
Closes #248. This PR replaces an [older, more convoluted, and hard to follow PR](https://github.com/filecoin-project/filecoin-docs/pull/252).

## This PR

- Creates an index page for `build/getting-started`.
- Splits `wallets-signing-tools-api-clients` into three separate pages.
- Moves `/build/developer-tools` and`/build/core-products` into `/build/tools`.
- Adds redirects for moved/deleted/renamed files.

### CODE REVIEW

**Comment by johnnymatthews** [comment]
In the [previous PR](https://github.com/filecoin-project/filecoin-docs/pull/252) I merged all the tutorials into single pages (instead of multiple pages). I've left that out of this PR to keep things tidy and easier to follow.

There were also some grammar, spelling, and formatting issues in the old PR that have not made it into this PR, again, for the same reasons.

File: docs/build/examples/sample-architectures.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ description: Learn how you might build an app on top of Filecoin.
 No matter what you wish to build on Filecoin, you'll generally need a few basic elements:
 
 - **A node.** Each Filecoin-based service or application will need to use at least one Filecoin node that maintains consensus. All interactions with the network must flow through an up-to-date node: sending and receiving market deals, sending and receiving data, and more. You can host this yourself, or choose a [hosted node](../start-building/interacting-with-the-network.md) option.
-- **An API.** Depending on language, and your preference for more granular or high-level commands, you can either directly use the node's go RPC API, or use one of the available [JS API client libraries](../developer-tools/wallets-signing-tools-api-clients.md#api-clients).
+- **An API.** Depending on language, and your preference for more granular or high-level commands, you can either directly use the node's go RPC API, or use one of the available [JS API client libraries](../tools/).
```
Comment (johnnymatthews) [inline docs/build/examples/sample-architectures.md:11]:
```suggestion
- **An API.** Depending on language, and your preference for more granular or high-level commands, you can either directly use the node's go RPC API, or use one of the available [JS API client libraries](../tools/).
```

@cwaring what do we do in situations like this, where the landing page for `/build/tools` is `README.md`. I imagine having this link as `../tools/README.md` would cause 404s to pop up on Netlify's end.

File: docs/build/start-building/implementations.md [hunk 2]
```diff
@@ -27,13 +27,15 @@ Here is a snapshot of each implementation’s progress across the primary parts
...
+**_We recommend that developers only build directly on Filecoin protocol implementations if they are very advanced developers or need to build at a very low level of the stack._** Filecoin protocol implementations expose low-level APIs that developers can use to integrate Filecoin directly into their applications. These APIs and their underlying functionality are still under heavy, active development, and require familiarity with blockchains and the Filecoin protocol. For most applications, we recommend developers use [FPS](../tools/filecoin-backed-pinning) or [Powergate](../tools/powergate).
```
Comment (johnnymatthews) [inline docs/build/start-building/implementations.md:34]:
```suggestion
**_We recommend that developers only build directly on Filecoin protocol implementations if they are very advanced developers or need to build at a very low level of the stack._** Filecoin protocol implementations expose low-level APIs that developers can use to integrate Filecoin directly into their applications. These APIs and their underlying functionality are still under heavy, active development, and require familiarity with blockchains and the Filecoin protocol. For most applications, we recommend developers use [FPS](../tools/filecoin-backed-pinning.md) or [Powergate](../tools/powergate.md).
```

File: docs/build/start-building/implementations.md [hunk 3]
```diff
@@ -27,13 +27,15 @@ Here is a snapshot of each implementation’s progress across the primary parts
...
+Generally speaking, you can use a Filecoin protocol implementation as a “client” (user) or a “miner” (storage provider) _(see [What is Filecoin?](../../introduction/what-is-filecoin) for more information)_. Most implementations separate out the storage miner functionality into a separate process from the core node functionality. The core node runs the blockchain system, makes storage and retrieval deals, performs data transfers, supports block producer logic, and syncs and validates the chain. The storage miner process produces sector commitments and _Proofs-of-Spacetime_ to prove they have been correctly storing storage client data, among other important functions. To be a Filecoin “client,” you need to run a full node. To be a Filecoin "miner," you must run a full node and the storage mining process.
```
Comment (johnnymatthews) [inline docs/build/start-building/implementations.md:38]:
```suggestion
Generally speaking, you can use a Filecoin protocol implementation as a “client” (user) or a “miner” (storage provider) _(see [What is Filecoin?](../../introduction/what-is-filecoin.md) for more information)_. Most implementations separate out the storage miner functionality into a separate process from the core node functionality. The core node runs the blockchain system, makes storage and retrieval deals, performs data transfers, supports block producer logic, and syncs and validates the chain. The storage miner process produces sector commitments and _Proofs-of-Spacetime_ to prove they have been correctly storing storage client data, among other important functions. To be a Filecoin “client,” you need to run a full node. To be a Filecoin "miner," you must run a full node and the storage mining process.
```

File: docs/build/start-building/implementations.md [hunk 4]
```diff
@@ -46,10 +48,10 @@ If interacting with the implementation’s RPC APIs, the general development pat
...
 
-Filecoin protocol implementations provide access to core Filecoin protocol workflows, with very few abstractions of the core concepts. This can be very useful, as it allows flexibility and precision in interacting with Filecoin network functionality. However, integrating nodes directly can require greater familiarity with how Filecoin works. To learn more about the various workflows enabled by the core Filecoin protocol, check-out the [How-tos](../../how-to/README.md).
+Filecoin protocol implementations provide access to core Filecoin protocol workflows, with very few abstractions of the core concepts. This can be very useful, as it allows flexibility and precision in interacting with Filecoin network functionality. However, integrating nodes directly can require greater familiarity with how Filecoin works.
 
 ::: tip
-See a walkthrough of an end-to-end example application (the Filecoin Network Inspector) built using [Lotus’ JS API client](../examples/network-inspector/overview.md).
+See a walkthrough of an end-to-end example application (the Filecoin Network Inspector) built using [Lotus’ JS API client](../examples/network-inspector/overview).
```
Comment (johnnymatthews) [inline docs/build/start-building/implementations.md:54]:
```suggestion
See a walkthrough of an end-to-end example application (the Filecoin Network Inspector) built using [Lotus’ JS API client](../examples/network-inspector/overview.md).
```

File: docs/build/start-building/interacting-with-the-network.md [hunk 5]
```diff
@@ -90,18 +89,19 @@ Here are some ways to spin up a mock version of the Filecoin network on your own
 
 ## Running Powergate
 
-You can also run an IPFS node + Filecoin lotus node + Powergate ([more info about Powergate](https://docs.filecoin.io/build/core-products/powergate/)) which enables using IPFS for hot storage and retrieval. For dapp developers who want some of the details of Filecoin's storage market abstracted as a convenience, Powergate also offers a storage deal agent, multi-tenant wallets mapped to the node's key (required to execute storage deals), automated deal renewal and other additional features. 
+You can also run an IPFS node + Filecoin lotus node + Powergate ([more info about Powergate](https://docs.filecoin.io/build/tools/powergate/)) which enables using IPFS for hot storage and retrieval. For dapp developers who want some of the details of Filecoin's storage market abstracted as a convenience, Powergate also offers a storage deal agent, multi-tenant wallets mapped to the node's key (required to execute storage deals), automated deal renewal and other additional features.
```
Comment (johnnymatthews) [inline docs/build/start-building/interacting-with-the-network.md:92]:
```suggestion
You can also run an IPFS node + Filecoin lotus node + Powergate ([more info about Powergate](https://docs.filecoin.io/build/tools/powergate.md)) which enables using IPFS for hot storage and retrieval. For dapp developers who want some of the details of Filecoin's storage market abstracted as a convenience, Powergate also offers a storage deal agent, multi-tenant wallets mapped to the node's key (required to execute storage deals), automated deal renewal and other additional features.
```

File: docs/build/tools/filecoin-backed-pinning.md [hunk 6]
```diff
@@ -1,64 +1,64 @@
...
 - Make storage requests through the API of the service you are using.
 - Maintain and monitor your data storage.
 - Retrieve your data through the API of the service you are using.
 
-This section will be updated with documentation links for additional FPS providers as they come online.
+This section will be updated with documentation links for additional FBP providers as they come online.
 
 ::: tip
-See a walkthrough of an end-to-end example application (the Meme Marketplace) built on Textile Buckets [here](../examples/meme-marketplace/overview.md).
+See a walkthrough of an end-to-end example application (the Meme Marketplace) built on Textile Buckets [here](../examples/meme-marketplace/overview).
```
Comment (johnnymatthews) [inline docs/build/tools/filecoin-backed-pinning.md:40]:
```suggestion
See a walkthrough of an end-to-end example application (the Meme Marketplace) built on Textile Buckets [here](../examples/meme-marketplace/overview.md).
```

File: docs/build/tools/powergate.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ From the [Textile docs](https://docs.textile.io/powergate/):
 > - Easily configure, connect, and deploy Powergate, [Lotus](https://lotu.sh), and [IPFS](https://ipfs.io) together.
 > - Much more!
 
-**_Powergate is the recommended solution for developers who want an easier interface and better performance from Filecoin, but who prefer to manage their own nodes._** For most developers, we recommend using an [FPS](./filecoin-backed-pinning-services.md). For very low-level or advanced developers, you can integrate [lotus](https://lotu.sh) directly into your application.
+**_Powergate is the recommended solution for developers who want an easier interface and better performance from Filecoin, but who prefer to manage their own nodes._** For most developers, we recommend using an [FPS](./filecoin-backed-pinning). For very low-level or advanced developers, you can integrate [lotus](https://lotu.sh) directly into your application.
```
Comment (johnnymatthews) [inline docs/build/tools/powergate.md:21]:
```suggestion
**_Powergate is the recommended solution for developers who want an easier interface and better performance from Filecoin, but who prefer to manage their own nodes._** For most developers, we recommend using an [FBP](./filecoin-backed-pinning.md). For very low-level or advanced developers, you can integrate [lotus](https://lotu.sh) directly into your application.
```

**Comment by cwaring** [comment]
@johnnymatthews are you intending to add content to the [Tools index page](https://deploy-preview-332--filecoin-docs.netlify.app/build/tools/)?

**Comment by johnnymatthews** [comment]
> @johnnymatthews are you intending to add content to the [Tools index page](https://deploy-preview-332--filecoin-docs.netlify.app/build/tools/)?

Yeahp. cb59e1d.

File: docs/build/tools/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+# Tools
+
+There are lots of tools developers can use to create Filecoin applications. While not every tool useful to Filecoin developers is listed here, these are the ones we think will help speed up the application development for new developers.
+
+## API clients
+
+API clients take care of the low-level details of making requests and handling responses and let you focus on writing code specific to your project. They can also translate between different programming languages. [Find out more →](./api-clients.md)
+
+## Filecoin-backed pinning
```
Comment (pooja) [inline docs/build/tools/README.md:14]:
```suggestion
## Filecoin-backed pinning services
```

File: docs/build/tools/README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+Filecoin-backed pinning services are data storage and retrieval services that offer the performance and availability of IPFS alongside the data persistence features of Filecoin’s decentralized storage network. [Find out more →](./filecoin-backed-pinning.md)
```
Comment (pooja) [inline docs/build/tools/README.md:16]: placeholder to rename file

File: docs/build/tools/filecoin-backed-pinning.md [hunk 10]
```diff
@@ -1,64 +1,64 @@
 ---
-title: Filecoin-backed IPFS Pinning Services
-description: A brief description of Filecoin-backed IPFS pinning services and how to use them.
+title: Filecoin-backed pinning
```
Comment (pooja) [inline docs/build/tools/filecoin-backed-pinning.md:2]: Please undo the renaming in this section back to the original and rename the file as well!

File: docs/build/tools/powergate.md [hunk 11]
```diff
@@ -18,7 +18,7 @@ From the [Textile docs](https://docs.textile.io/powergate/):
 > - Easily configure, connect, and deploy Powergate, [Lotus](https://lotu.sh), and [IPFS](https://ipfs.io) together.
 > - Much more!
 
-**_Powergate is the recommended solution for developers who want an easier interface and better performance from Filecoin, but who prefer to manage their own nodes._** For most developers, we recommend using an [FPS](./filecoin-backed-pinning-services.md). For very low-level or advanced developers, you can integrate [lotus](https://lotu.sh) directly into your application.
+**_Powergate is the recommended solution for developers who want an easier interface and better performance from Filecoin, but who prefer to manage their own nodes._** For most developers, we recommend using an [FBP](./filecoin-backed-pinning.md). For very low-level or advanced developers, you can integrate [lotus](https://lotu.sh) directly into your application.
```
Comment (pooja) [inline docs/build/tools/powergate.md:21]:
```suggestion
**_Powergate is the recommended solution for developers who want an easier interface and better performance from Filecoin, but who prefer to manage their own nodes._** For most developers, we recommend using an [FPS](./filecoin-backed-pinning-services.md). For very low-level or advanced developers, you can integrate [lotus](https://lotu.sh) directly into your application.
```

File: docs/build/tools/signing-libraries.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+---
+title: Signing libraries
+description:
```
Comment (pooja) [inline docs/build/tools/signing-libraries.md:3]:
```suggestion
description: Signing libraries can be used to sign transactions without requiring a dedicated node.
```

File: docs/introduction/faq.md [hunk 13]
```diff
@@ -186,4 +186,4 @@ Also, note that world location matters for mining: many clients will prefer mine
...
 
-If you are retrieving your data from IPFS or a remote pinning layer such as an [FPS](../build/core-products/filecoin-backed-pinning-services.md), retrieval should take on the order of milliseconds to seconds in the worst case. Our latest tests for retrieval from the Filecoin network directly show that a sealed sector holding data takes ~1 hour to unseal. 1-5 hours is our best real world estimate to go from sector unsealing to ultimate delivery of the data. If you need faster data retrieval for your application, we recommend building on Powergate or an FPS.
+If you are retrieving your data from IPFS or a remote pinning layer such as an [FPS](../build/tools/filecoin-backed-pinning.md), retrieval should take on the order of milliseconds to seconds in the worst case. Our latest tests for retrieval from the Filecoin network directly show that a sealed sector holding data takes ~1 hour to unseal. 1-5 hours is our best real world estimate to go from sector unsealing to ultimate delivery of the data. If you need faster data retrieval for your application, we recommend building on Powergate or an FPS.
```
Comment (pooja) [inline docs/introduction/faq.md:189]: placeholder to rename file

File: docs/build/start-building/interacting-with-the-network.md [hunk 14]
```diff
@@ -105,9 +105,9 @@ To tap the faucet or see dashboard for Nerpa:
 - Other Nerpa Devnet Info: http://www.nerpa.fildev.network/
 - For updates see: [#fil-net-nerpa](https://filecoinproject.slack.com/archives/C016VJSJNTH) (in Filecoin Slack)
 
-
 ### Calibration Devnet
```
[the remark this answers is not in the document]
Comment (johnnymatthews) [reply docs/build/start-building/interacting-with-the-network.md:108]: Sounds we remove all references to _Calibration devnet_ then?
